presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorSession.java

```diff
@@ -51,4 +52,6 @@ default long getStartTime()
     Instant getStart();
 
     <T> T getProperty(String name, Class<T> type);
+
+    Set<String> getPropertiesNames();
```

Comment:
This is something we wanted to avoid, to modify SPI for this.

Reply:
Do you have an idea how to inject Session properties to CachingJdbcClient ?

Reply:
In the same way they are provided to `JdbcConnector`, with `Set<SessionPropertiesProvider> sessionProperties`